lib/Auth/Process/checkClientIP.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * This Auth Proc Filter allows to check the IP of the client and to control the 
+ * two-factor authentication of the following privacyIDEA Auth Proc Filter depending on the client IP.
+ * For example, it can be used to configure that a user does not need to provide a 
+ * second factor when logging in from the local network.
+ * @author Micha Preußer <micha.preusser@netknights.it>
+ */
+
+class sspmod_privacyIDEA_Auth_Process_checkClientIP extends SimpleSAML_Auth_ProcessingFilter {
+
+	/**
+	 * enter excluded ip addresses
+	 * @var array|mixed
+	 */
+     private $excludeClientIPs = array();
+
+     public function __construct(array $config, $reserved)
+     {
+        SimpleSAML_Logger::info("Checking client ip for privacyIDEA");
+        parent::__construct($config, $reserved);
+        $cfg = SimpleSAML_Configuration::loadFromArray($config, 'privacyidea:checkClientIP');
+        $this->excludeClientIPs = $cfg->getArray('excludeClientIPs', null);
+     }
+
+	public function process( &$state ) {
+		$clientIP = ip2long($_SERVER['REMOTE_ADDR']);
```

Comment:
We should also evaluate X-Forwarded-For and HTTP_X_FORWARDED_FOR.

Reply:
https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/44/commits/6ad4ae71ccd5a179ab6b1ff0e20d0ef76fe1036b should fix it.